uaclient/entitlements/base.py

```diff
@@ -590,10 +583,8 @@ def applicability_status(
                 )
                 return (ApplicabilityStatus.INAPPLICABLE, invalid_msg)
 
-            if not match:
-                return ApplicabilityStatus.INAPPLICABLE, invalid_msg
-            kernel_major = int(match.group("major"))
-            kernel_minor = int(match.group("minor"))
+            kernel_major = int(kernel_info.major)
```

Comment:
Can't we make that type casting directly when creating the `KernelInfo` object ?

Reply:
Yep good idea, we can do this for major minor and patch at least